js/DOMElementModifications.js

```diff
@@ -0,0 +1,287 @@
+      // Collect all possible nodes
+      const targetNodes = isChangedEvent
+        ? [ item.target ]
+        : Array.from(isAddedEvent ? addedNodes : removedNodes)
+          .flatMap(node => {
+            return this._watch.subtree
+              // Fetch all decendant nodes
+              ? [node, ...node.querySelectorAll('*')]
+              // Use only immediate node
+              : [node];
+          });
+      // Fetch initial event name
+      const eventName = isAddedEvent
+        ? 'added'
+        : isRemovedEvent
+          ? 'removed'
+          : 'changed';
+      targetNodes.forEach(target => {
+        // Keep changes for each node
+        const value = changes.get(target) ?? {
+          eventNames: [],
+          previousAttributes: this._watch.attributes
+            ? {}
+            : null
+        };
+        value.eventNames.push(eventName);
+        // Collect changed attributes
+        const shouldRecordAttribute = (isChangedEvent && item.attributeName);
+        if (shouldRecordAttribute) value.previousAttributes[item.attributeName] ??= item.oldValue;
+        changes.set(target, value);
+      });
+      return changes;
+    }, new Map());
+    for (let [target, { eventNames, previousAttributes }] of reducedChanges.entries()) {
+      // Check if the added and removed events are a moved node, which should have a changed event
+      eventNames = DOMElementModifications.calculateEventNames(eventNames);
+      eventNames.forEach(eventName => {
+        const isNodeAdded = (eventName === 'added');
+        const isNodeRemoved = (eventName === 'removed');
+        // Find events to trigger, filter by and return event name selectors
+        const selectorFilters = isNodeAdded
+          ? this._addedFilters
+          : isNodeRemoved
+            ? this._removedFilters
+            : this._changedFilters;
+        if (!selectorFilters.length) return;
+        const selectors = selectorFilters.map(filter => filter(target)).filter(Boolean);
+        if (!selectors.length) return;
+        // Capture new attribute values
+        const changedAttributes = this._watch.attributes
+          ? Object.keys(previousAttributes).reduce((attributes, attrName) => {
+            attributes[attrName] = target.getAttribute(attrName);
+            return attributes;
+          }, {})
+          : null;
+        // Trigger events.
+        // Always trigger a changed event, trigger added or removed only when specifically requested
+        const EventObject = new DOMElementModificationEventObject({
+          type: eventName,
+          target,
+          changedAttributes,
+          previousAttributes
+        });
+        selectors.forEach(selector => {
+          if (selector && selector !== true) {
+            // Event attachment has specified selector
+            if (eventName !== 'changed') this.trigger(`${eventName}:${selector}`, EventObject);
```

Comment:
I think the if check can be removed and line 193 deleted as they both achieve the same thing

Reply:
192 does the `added`/`removed` events (when an explicit listen has been attached), 193 does the `changed` events, either subsequent to the `added`/`removed one, or on its own, always.

Reply:
ah good point. My mistake, sorry. I had assumed a `return` that wasn't actually there 👍